jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py

```diff
@@ -110,15 +116,47 @@ def _set_default_results_label(self, event={}):
         label_comps += [f"moment {self.n_moment}"]
         self.results_label_default = " ".join(label_comps)
 
-    @observe("dataset_selected")
+    @observe("dataset_selected", "n_moment")
     def _set_data_units(self, event={}):
+
+        if self.n_moment == 0:
+            self.output_unit_selected = "Flux"
+        elif ((self.n_moment == 1 and self.output_unit_selected in ("Flux", "Velocity^N")) or
+              (self.n_moment > 1 and self.output_unit_selected in ("Flux", "Wavelength"))):
+            self.output_unit_selected = "Velocity"
+        self.send_state("output_unit_selected")
+
+        unit_dict = {"Flux": "",
+                     "Wavelength": "",
+                     "Velocity": "km/s",
+                     "Velocity^N": f"km{self.n_moment}/s{self.n_moment}"}
+
         if self.dataset_selected != "":
             # Spectral axis is first in this list
+            data = self.app.data_collection[self.dataset_selected]
             if self.app.data_collection[self.dataset_selected].coords is not None:
-                unit = self.app.data_collection[self.dataset_selected].coords.world_axis_units[0]
-                self.dataset_spectral_unit = unit
+                sunit = data.coords.world_axis_units[0]
+                self.dataset_spectral_unit = sunit
+                unit_dict["Wavelength"] = sunit
             else:
                 self.dataset_spectral_unit = ""
+            unit_dict["Flux"] = data.get_component('flux').units
+
+        # Update units in selection item dictionary
+        for item in self.output_unit_items:
+            item["unit_str"] = unit_dict[item["label"]]
+
+        # Filter what we want based on n_moment
+        if self.n_moment == 0:
+            self.output_radio_items = [self.output_unit_items[0],]
+        elif self.n_moment == 1:
+            self.output_radio_items = self.output_unit_items[1:3]
+        else:
+            self.output_radio_items = self.output_unit_items[2:]
```

Comment:
how does this look to the user using the API?

Reply:
The user still sees the full options list in the API.

Reply:
If anything isn't valid, we should raise errors in `calculate_moment` then?